src/backend/utils/misc/queryenvironment.c

```diff
@@ -435,25 +435,27 @@ bool ENRgetSystableScan(Relation rel, Oid indexId, int nkeys, ScanKey key, List
 				* Search through the entire ENR relation list for everything
 				* that has a relation (non-recursive) to this object.
 				* If indexId is DependDependerIndexId, we try to mimic
-				* SELECT * FROM pg_depend WHERE classid=v1 AND objid=v2
+				* SELECT * FROM pg_depend WHERE classid=v1 AND objid=v2 (AND objsubid = v3 if applicable)
 				* Otherwise if it is DependReferenceIndexId we try to mimic
-				* SELECT * FROM pg_depend WHERE refclassid=v1 AND refobjid=v2
+				* SELECT * FROM pg_depend WHERE refclassid=v1 AND refobjid=v2 (AND refobjsubid = v3 if applicable)
 				* So we cannot return right away if there is a match.
 				*/
 				ListCell   *lc;
 				foreach(lc, enr->md.cattups[ENR_CATTUP_DEPEND]) {
 					Form_pg_depend tup = (Form_pg_depend) GETSTRUCT((HeapTuple) lfirst(lc));
 					if (indexId == DependDependerIndexId &&
 						tup->classid == (Oid)v1 &&
-						tup->objid == (Oid)v2)
+						tup->objid == (Oid)v2 &&
+						(nkeys == 2 || tup->objsubid == (int32)v3))
```

Comment:
Why other have (Oid) to cast to Oid, while v3 is cast to (int32) , and v1,v2,v3 is really confusing in naming, should assign a more reasonable naming.

Reply:
As above, this is cast to int32 in order to match the definition of the field in ForData_pg_depend. We use generic v1, v2, v3 naming as they are used to refer to the keys being passed into the function, which could be for any catalog table, not just pg_depend. Thus, the value they represent can be multiple different types depending on which catalog we are searching.